src/test/java/com/bobocode/hoverla/bring/context/BeanValidatorTest.java

```diff
@@ -0,0 +1,160 @@
+        beanDefinitionList.add(beanDefinition);
+        beanDefinitionList.add(beanDefinitionWithConflictDependencies);
+
+        assertExceptionAndMessage(expectedMessage);
+    }
+
+    @Test
+    @DisplayName("Fails on dependency validation when the found dependency type is not suitable for the required type")
+    void unsuitableTypeForDependency() {
+        String expectedMessage = "Types are not coincide for required type";
+        BeanDefinition beanDefinition = prepareDefinition("anotherType", String.class);
+        BeanDefinition beanDefinitionWithConflictDependencies = prepareDefinition("BD5", String.class,
+                beanDefinition);
+        beanDefinitionList.add(beanDefinition);
+        beanDefinitionList.add(beanDefinitionWithConflictDependencies);
+
+        doReturn(Integer.class).when(beanDefinition).type();
+
+        assertExceptionAndMessage(expectedMessage);
+    }
+
+    @Test
+    @DisplayName("Fails when beans have a circular dependency")
+    void circularDependency() {
+        String expectedMessage = "the last one depends on first";
+        BeanDefinition beanDefinition4 = beanDefinitionList.get(3);
+        BeanDefinition beanDefinition3 = beanDefinitionList.get(2);
+        beanDefinition4.dependencies().put(beanDefinition3.name(), beanDefinition3.type());
+
+        assertExceptionAndMessage(expectedMessage);
+    }
+
+    @Test
+    @DisplayName("Fails when didn't find dependency by name, but found by type more than 1")
+    void moreThanOneDependencyByType() {
+        String expectedMessage = "Found more than 1 beans with type";
+        BeanDefinition beanDefinition = prepareDefinition("beanDef5", Integer.class);
+        beanDefinition.dependencies().put("anotherName", Integer.class);
+        beanDefinitionList.add(beanDefinition);
+
+        assertExceptionAndMessage(expectedMessage);
+    }
+
+    @Test
+    @DisplayName("Fails when didn't find dependency by name and type")
+    void dependenciesNotFound() {
+        String expectedMessage = "No found beans with type";
+        BeanDefinition beanDefinition = prepareDefinition("beanDef5", Integer.class);
+        beanDefinition.dependencies().put("anotherName", BeanDefinition.class);
+        beanDefinitionList.add(beanDefinition);
+
+        assertExceptionAndMessage(expectedMessage);
+    }
+
+    @Test
+    @DisplayName("Fails when context contains beans with duplicate names")
+    void duplicateNames() {
+        String expectedMessage = "Context contains beans with duplicate names: ";
+        BeanDefinition beanDefinition = prepareDefinition(BD1, String.class);
+        beanDefinitionList.add(beanDefinition);
+
+        assertExceptionAndMessage(expectedMessage);
+    }
+
+    private void assertExceptionAndMessage(String expectedMessage) {
+        var ex = Assertions.assertThrows(BeanValidationException.class, () -> beanValidator.validate(beanDefinitionList));
+        Assertions.assertTrue(ex.getMessage().contains(expectedMessage));
```

Comment:
The same, why do you use contains instead of checking for equality

Reply:
I'd say that it'd better if you setup the list of beans for each test case separately, because it seems like you have only one case which uses them all. Then, it'd better to have this guy look like this
`    private void assertExceptionAndMessage(String expectedMessage, Runnable runnable) {
`    
   so that for each test case I see a section // WHEN, when you actually call a code that we test

Reply:
done